js/plugins/cloud-sql-pg/src/index.ts

```diff
@@ -127,26 +144,158 @@ export function configurePostgresRetriever<
  * @param ai A Genkit instance
  * @param params The params for the indexer
  * @param params.tableName The name of the indexer
+ * @param params.engine The engine to use for the indexer
  * @param params.embedder The embedder to use for the retriever
  * @param params.embedderOptions  Options to customize the embedder
- * @returns A Genkit indexer
+ * @param params.metadataColumns The metadata columns to use for the indexer
+ * @param params.idColumn The id column to use for the indexer
+ * @param params.metadataJsonColumn The metadata json column to use for the indexer
+ * @param params.distanceStrategy The distance strategy to use for the indexer
+ * @param params.contentColumn The content column to use for the indexer
+ * @param params.embeddingColumn The embedding column to use for the indexer
+ * @param params.schemaName The schema name to use for the indexer
+ * @param params.chunkSize The chunk size to use for the indexer
+ * @returns Add documents to vector store
+ * 
+ * 
+ * 
  */
 export function configurePostgresIndexer<
   EmbedderCustomOptions extends z.ZodTypeAny,
 >(
   ai: Genkit,
   params: {
     tableName: string;
+    engine: PostgresEngine;
+    schemaName?: string;
+    contentColumn: string;
+    embeddingColumn: string;
+    metadataColumns?: string[];
+    idColumn: string;
+    metadataJsonColumn?: string;
+    distanceStrategy: 'cosine' | 'ip' | 'l2';
     embedder: EmbedderArgument<EmbedderCustomOptions>;
     embedderOptions?: z.infer<EmbedderCustomOptions>;
   }
 ) {
+  if (!params.engine) {
+    throw new Error('Engine is required');
+  }
+
+  const {
+    tableName,
+    engine,
+    schemaName = 'public',
+    contentColumn,
+    embeddingColumn,
+    metadataColumns = [],
+    idColumn,
+    metadataJsonColumn,
+    distanceStrategy,
+    embedder,
+    embedderOptions
+  } = params;
+
+  async function ensureTableExists() {
+    await engine.pool.raw('CREATE EXTENSION IF NOT EXISTS vector');
+    
+    await engine.pool.raw(`
+      CREATE TABLE IF NOT EXISTS ${schemaName}.${tableName} (
+        ${idColumn} UUID PRIMARY KEY,
+        ${contentColumn} TEXT NOT NULL,
+        ${embeddingColumn} VECTOR(1536),
+        ${metadataJsonColumn ? `${metadataJsonColumn} JSONB,` : ''}
+        ${metadataColumns.map(col => `${col} TEXT,`).join('\n')}
+        created_at TIMESTAMP DEFAULT NOW()
+      );
+      
+      CREATE INDEX IF NOT EXISTS idx_${tableName}_embedding 
+      ON ${schemaName}.${tableName} 
+      USING ivfflat (${embeddingColumn}) 
+      WITH (lists = 100);
+    `);
+  }
```

Comment:
Table creation is handled by the init table methods in the engine class. Use the langchain constructor to check that table columns exist